hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/WALEntryStream.java

```diff
@@ -249,7 +249,7 @@ private boolean checkAllBytesParsed() throws IOException {
             + ", which is too far away from reported file length " + stat.getLen()
             + ". Restarting WAL reading (see HBASE-15983 for details). " + getCurrentPathStat());
         setPosition(0);
-        resetReader();
+        openReader(currentPath, true);
```

Comment:
How about add a resetCompressionContext() method in the implementation of Reader.reset()? That means when the reader is reset, the compressionContext needs also be reset.
And then the resetReader() can be kept everywhere?

Reply:
@sunhelly Pls see [WALEntryStream#resetReader](https://github.com/apache/hbase/blob/branch-1/hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/WALEntryStream.java#L391)
There will call _seek_ method, which means the resetReader also might reposition the cursor somewhere, so i think we cannot just clear the compressionContext.
As a temporary/staged fix, I think creating the reader directly is the least invasive... What do you think?